src/main/java/org/datadog/jenkins/plugins/datadog/clients/DogStatsDClient.java

```diff
@@ -334,6 +334,13 @@ public boolean sendLogs(String payload) {
         try {
             this.ddLogger.info(payload);
 
+            // Check if we have handlers in our logger. This may happen when ddLogger initialization fails
+            // ddLogger may not be null but may be mis-configured.
+            // Reset to null to reinitialize if needed.
+            if(this.ddLogger.getHandlers().length == 0){
```

Comment:
Nit: Could we move this up to the `if(!status)` line since we're checking the health of the logger there anyway? 

Thinking out loud: Whether this code is here or up outside this try block, it looks like we'll lose this log line if the logger isn't initialized properly. Which seems fair.